src/main/java/org/traccar/geocoder/PhotonGeocoder.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2014 - 2024 Anton Tananaev (anton@traccar.org)
+ * Copyright 2024 - 2024 Matjaž Črnko (m.crnko@txt.i)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.geocoder;
+
+import jakarta.json.JsonArray;
+import jakarta.json.JsonObject;
+import jakarta.ws.rs.client.Client;
+
+public class PhotonGeocoder extends JsonGeocoder {
+
+    private static String formatUrl(String url, String key, String language) {
+        if (url == null) {
+            url = "https://photon.komoot.io/reverse";
+        }
+        url += "?lat=%f&lon=%f";
+        if (key != null) {
+            url += "&key=" + key;
+        }
+        if (language != null) {
+            url += "&lang=" + language;
+        }
+        return url;
+    }
+
+    public PhotonGeocoder(
+            Client client, String url, String key, String language, int cacheSize, AddressFormat addressFormat) {
+        super(client, formatUrl(url, key, language), cacheSize, addressFormat);
+    }
+
+    @Override
+    public Address parseAddress(JsonObject json) {
```

Comment:
The format seems to be similar to `BanGeocoder`. Can we combine the two?

Reply:
Oh damn, I actually flicked thorough some of the Geocoders, looking if anything similar is supported but did not see the Ban one unfortunately. This seems to be GeoJSON 'standard' format (as Nominatim also has this format option).
The only clash I see is the `name` key, which contains different type of data. But we could differentiate this by `osm_id` key (as it looks like the only such difference). I will take a deeper look in a few days and get back to you if I find any other 'ugly' differences.

Reply:
Ok, I did some more research and the `name` difference is avoidable (as we get separate data, and can store that). The only remaining problem is that Ban does not return the country. So I was thinking of creating a [GeocodeJSON](https://github.com/geocoders/geocodejson-spec) geocoder that can be used with the spec adhering geocoders and then a rework of Ban geocoder that extends GeocodeJSONGeocoder and just 'adds' the hardcoded url & countrycode. What do you think?